contracts/interfaces/zap/ISolidZapCollateralize.sol

```diff
@@ -10,4 +10,54 @@ interface ISolidZapCollateralize {
     function swManager() external view returns (address);
 
     function forwardContractBatch() external view returns (address);
+
+    /// @notice Zap function that achieves the following:
+    /// 1. Transfers `amountIn` forward credits with batch id `batchId` to this contract
+    /// 2. Collateralizes the transferred forward credits via SolidWorldManager, receives `crispToken`
+    /// 3. Swaps `crispToken` for `outputToken` via encoded swap
+    /// 4. Transfers the `outputToken` balance of SolidZapCollateralize to `msg.sender`
+    /// 5. Transfers remaining `crispToken` balance of SolidZapCollateralize to the `dustReceiver`
+    /// @notice The `msg.sender` must approve this contract to spend the forward credits
+    /// @param outputToken The token used for obtaining forward credits
+    /// @param crispToken The intermediate token used for redeeming forward credits
+    /// @param batchId The batch id of the forward credits to collateralize
+    /// @param amountIn The amounts of forward credits to collateralize
+    /// @param amountOutMin The minimum amounts of `crispToken` to receive from collateralization
+    /// @param swap Encoded swap from `crispToken` to `outputToken`
+    /// @param dustReceiver Address to receive any remaining `crispToken` dust
+    function zapCollateralize(
+        address outputToken,
+        address crispToken,
+        uint batchId,
+        uint amountIn,
+        uint amountOutMin,
+        bytes calldata swap,
```

Comment:
Amount out is already encoded in swap operation. So is it possible to get rid of `amountOutMin` in this? Will it make sense?

Reply:
it's in the swap but it's actually a pain to extract, and it totally depends on what encoded swap we receive from 1inch / 0x. 

For example, 1inch router has `uniswapV3Swap`, `swap`, `clipperSwap`, `unoswap`. To extract the `amountOutMin` from the encoded swap, first you need to know what kind of encoded swap you are dealing with, then know the signature, then extract the amountOutMin from there. We don't want to deal with that in our zap contract

Reply:
What if we set amountOutMin to zero? The transaction will fail if swap can't be done, right?

Reply:
yes, the tx fails as soon as something inside it fails

`amountOutMin` is relevant only for our `manager.collateralizeBatch` call, it can be 0 and it just means one less check

the swap on the other hand which goes from crisp-m to X has to be encoded correctly with the amount of crisp-m expected from the collateralization (which we'll do in our frontend)

Reply:
So we anyway operate with `amountOutMin`.

Reply:
ofc